File: python-service/app/schemas/risk.py
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.common import InternalRequestMeta


class PrepayRiskOrderDTO(BaseModel):
    id: str
    buyer_user_id: str
    seller_user_id: str
    amount_fen: int
    service_type: str
    district: str | None = None


class PrepayRiskPaymentDTO(BaseModel):
    channel: Literal["alipay", "wechat_pay"]
    scene: Literal["deposit", "full", "balance"]


class PrepayRiskCheckRequest(BaseModel):
    meta: InternalRequestMeta
    order: PrepayRiskOrderDTO
    payment: PrepayRiskPaymentDTO


class PrepayRiskCheckResponse(BaseModel):
    request_id: str
    allowed: bool
    risk_level: Literal["low", "medium", "high"]
    reason_codes: list[str] = Field(default_factory=list)
    human_message: str | None = None

File: server/src/internal-dto/risk.ts
import type { PaymentChannel, PaymentScene, RiskLevel } from '@mimi/shared';

import type { InternalRequestMeta } from './common.js';

export interface PrepayRiskCheckRequest {
  meta: InternalRequestMeta;
  order: {
    id: string;
    buyerUserId: string;
    sellerUserId: string;
    amountFen: number;
    serviceType: string;
    district?: string;
  };
  payment: {
    channel: PaymentChannel;
    scene: PaymentScene;
  };
}

export interface PrepayRiskCheckResponse {
  requestId: string;
  allowed: boolean;
  riskLevel: RiskLevel;
  reasonCodes: string[];
  humanMessage?: string;
}
